pkg/trace/stats/concentrator.go

```diff
@@ -70,6 +88,9 @@ func NewConcentrator(conf *config.AgentConfig, out chan *pb.StatsPayload, now ti
 		peerSvcAggregation:     conf.PeerServiceAggregation,
 		computeStatsBySpanKind: conf.ComputeStatsBySpanKind,
 	}
+	if conf.PeerServiceAggregation && len(conf.PeerTags) != 0 {
```

Comment:
You already check the length of `conf.PeerTags` in `prepareTagKeys`, so no need to do it twice.